src/zod-collection.ts

```diff
@@ -21,7 +21,18 @@ export const convertToZodCollection = <TSchema extends Document>(
     preInsert: (
       obj: OptionalUnlessRequiredId<TSchema>
     ): OptionalUnlessRequiredId<TSchema> => {
-      return schema.parse(obj) as OptionalUnlessRequiredId<TSchema>
+      const schemaWithId = schema.and(
+        z.object({
+          _id: z
+            // Can't be instanceof(ObjectId) because the instanceof operator
+            // doesn't work correctly when ts-mongo is imported by another lib
+            // This is similar to what we do with zod instance checking
```

Comment:
Similar to what we do with zod instance checking, where?

Reply:
Here, in this `zod` fork that is maintained in my GitHub account: https://github.com/cau777/zod/commit/ee7f929f6b145721e5f79ad8ba7a2357d32053f6

We use it instead of the official version of `zod`: https://github.com/cau777/ts-mongo/blob/29ca813679e62731df514b97f91d18d773155930/package.json#L51-L51

Reply:
Let's add this permalink as a comment here, to avoid any confusions:
https://github.com/cau777/zod/blob/ee7f929f6b145721e5f79ad8ba7a2357d32053f6/src/types.ts#L206-L226